src/zope/app/apidoc/static.py

```diff
@@ -159,10 +159,23 @@ def setUserAndPassword(self, user, pw):
         self.addHeader('Authorization', 'Basic %s:%s' % (user, pw))
 
     @classmethod
-    def begin(cls):
-        # TODO: We need to let this define what config file to execute.
+    def begin(cls, package_name='', zcml_file='configure.zcml'):
         from zope.app.apidoc.testing import APIDocLayer
         from zope.app.appsetup import appsetup
+
+        if package_name:
+            from zope.configuration import xmlconfig
+            xmlconfig.file('ftesting.zcml', zope.app.apidoc)
+
+            import importlib
+            package = importlib.import_module(package_name)
+            APIDocLayer.__init__(
```

Comment:
Calling `__init__` on an existing object is quite strange. Could you just assign `APIDocLayer = testing.BrowserLayer(package, name=...)`? Or is there some reason the change has to be done to this exact object?

Reply:
I'm still quite unfamiliar with APIDocLayer's side effects, so I was just trying to make sure that changes done to APIDocLayer would persist outside of `PublisherBrowser.begin` in order to maintain predictability.

Some recent experimentation on my part suggests that this is unnecessary.

Reply:
Even worse, using `__init__` ended up breaking tests.